src/datasources/account/account.datasource.ts

```diff
@@ -0,0 +1,294 @@
+import { Inject, Injectable } from '@nestjs/common';
+import * as postgres from 'postgres';
+import { IAccountDataSource } from '@/domain/interfaces/account.datasource.interface';
+import { AccountDoesNotExistError } from '@/datasources/account/errors/account-does-not-exist.error';
+import {
+  Account as DomainAccount,
+  EmailAddress,
+} from '@/domain/account/entities/account.entity';
+import { Subscription as DomainSubscription } from '@/domain/account/entities/subscription.entity';
+
+interface Account {
+  id: number;
+  chain_id: number;
+  email_address: string;
+  safe_address: string;
+  account: string;
+  verified: boolean;
+  verification_code: string | null;
+  verification_code_generated_on: Date | null;
+  verification_sent_on: Date | null;
+  unsubscription_token: string;
+}
+
+interface Subscription {
+  id: number;
+  key: string;
+  name: string;
+}
+
+@Injectable()
+export class AccountDataSource implements IAccountDataSource {
+  constructor(@Inject('DB_INSTANCE') private readonly sql: postgres.Sql) {}
+
+  async getVerifiedAccountEmailsBySafeAddress(args: {
+    chainId: string;
+    safeAddress: string;
+  }): Promise<{ email: string }[]> {
+    const emails = await this.sql<{ email_address: string }[]>`
+        SELECT email_address
+        FROM accounts
+        WHERE chain_id = ${args.chainId}
+          AND safe_address = ${args.safeAddress}
+          AND verified is true
+        ORDER by id`;
+
+    return emails.map((email) => ({ email: email.email_address }));
+  }
+
+  async getAccount(args: {
+    chainId: string;
+    safeAddress: string;
+    account: string;
+  }): Promise<DomainAccount> {
+    const [account] = await this.sql<Account[]>`SELECT *
+                                              FROM accounts
+                                              WHERE chain_id = ${args.chainId}
+                                                AND safe_address = ${args.safeAddress}
+                                                AND account = ${args.account}`;
+    if (!account) {
+      throw new AccountDoesNotExistError(
+        args.chainId,
+        args.safeAddress,
+        args.account,
+      );
+    }
+
+    return {
+      chainId: account.chain_id.toString(),
+      emailAddress: new EmailAddress(account.email_address),
+      isVerified: account.verified,
+      safeAddress: account.safe_address,
+      account: account.account,
+      verificationCode: account.verification_code,
+      verificationGeneratedOn: account.verification_code_generated_on,
+      verificationSentOn: account.verification_sent_on,
+    };
+  }
+
+  async saveAccount(args: {
```

Comment:
Nit: what do you think about renaming (and propagating) this to `createAccount` as this is inserting, not updating.

Reply:
[fa63a4e](https://github.com/safe-global/safe-client-gateway/pull/1050/commits/fa63a4e9950036c1b044c936774359b805c7c62f)